Connorwon 222 alumni vs member separation

ConnorWon opened the pull request:
## Description

Backend portion of the alumni vs non-alumni member separation

### Changes

* Added new member endpoint which gets members apart of a specific project
* Got rid of FrontEndAPI checking in get request member controllers
* Updated member model to include an isAlumni field and got rid of redundant 'Message' field
* Added and updated member controller tests based on changes

### Updates

[Please enter a brief summary of updates to this PR during code review - if any]

## Checklist

- [x] Ran `npm run lint:fix` to format code
- [x] Assigned PR to all authors
- [x] Requested at least one reviewer

ConnorWon commented:
![](https://github.trello.services/images/mini-trello-icon.png) [project members vs alumni separation](https://trello.com/c/KxCNxlPq/222-project-members-vs-alumni-separation)

ConnorWon commented:
![](https://github.trello.services/images/mini-trello-icon.png) [project members vs alumni separation](https://trello.com/c/KxCNxlPq/222-project-members-vs-alumni-separation)

ElisaHu1 commented on test/unit/controllers/memberControllerTest.js:
I LOVE you adding a unit test for this!

ryanyae reviewed (changes requested):
Just as a safe measure we should just add some dummy data to show how it would look like in prod and before we merge this in we should just delete the dummy data

ConnorWon commented:
> Just as a safe measure we should just add some dummy data to show how it would look like in prod and before we merge this in we should just delete the dummy data

Attached two videos, each showing what the project and people page would look like with dummy data included

https://github.com/UBC-VCL/VCL-content-platform-server/assets/120760211/a77edf24-263b-4b05-8b74-e66ae6e93e47


https://github.com/UBC-VCL/VCL-content-platform-server/assets/120760211/5a631b8a-ab89-41b3-9819-4b23211ed4bf

ryanyae reviewed (changes requested):
Looks good, @ElisaHu1 I really like Connor's implementation of the team members better than that of Jasmine's. I think we should merge this ticket in and close Jasmine's ticket regarding the team members mobile view ticket. @ConnorWon I would also like to add that we dont need the breadcrumbs at the top anymore, do you also think you could add a max-width property to the alumni button (image for reference below).

![image](https://github.com/UBC-VCL/VCL-content-platform-server/assets/76890786/68982cc8-ae04-4017-b7c9-8414ce522cb3)

ConnorWon commented:
> Looks good, @ElisaHu1 I really like Connor's implementation of the team members better than that of Jasmine's. I think we should merge this ticket in and close Jasmine's ticket regarding the team members mobile view ticket. @ConnorWon I would also like to add that we dont need the breadcrumbs at the top anymore, do you also think you could add a max-width property to the alumni button (image for reference below).
> 
> EDIT: I realize this is the backend PR, but pretend like this request is for the frontend PR.
> 
> ![image](https://private-user-images.githubusercontent.com/76890786/316169065-68982cc8-ae04-4017-b7c9-8414ce522cb3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTEyMzYwOTksIm5iZiI6MTcxMTIzNTc5OSwicGF0aCI6Ii83Njg5MDc4Ni8zMTYxNjkwNjUtNjg5ODJjYzgtYWUwNC00MDE3LWI3YzktODQxNGNlNTIyY2IzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzIzVDIzMTYzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY3NmYwZTg3NmZkOTQ2ZTJhNWFiNmY1MmRmMWE1MWM1MTI1Mzc1OTVlODU5MWUxMWFmNWJlMDM0ZGEzMGQ2NmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.omNoOPE241SHHh70KYAxwQKdCY4GVK5lbnH3kbCBSrw)

Removed the breadcrumbs and added the max-width

ElisaHu1 commented:
> Looks good, @ElisaHu1 I really like Connor's implementation of the team members better than that of Jasmine's. I think we should merge this ticket in and close Jasmine's ticket regarding the team members mobile view ticket. @ConnorWon I would also like to add that we dont need the breadcrumbs at the top anymore, do you also think you could add a max-width property to the alumni button (image for reference below).
> 
> EDIT: I realize this is the backend PR, but pretend like this request is for the frontend PR.
> 
> ![image](https://private-user-images.githubusercontent.com/76890786/316169065-68982cc8-ae04-4017-b7c9-8414ce522cb3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTEzMzIyNDUsIm5iZiI6MTcxMTMzMTk0NSwicGF0aCI6Ii83Njg5MDc4Ni8zMTYxNjkwNjUtNjg5ODJjYzgtYWUwNC00MDE3LWI3YzktODQxNGNlNTIyY2IzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI1VDAxNTkwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI4NjdlNmJkNmZiOTFmNjU3NmQ2ZGU3ODdlZWMyMmI1MTA1NDQ0M2I2NDM3ZWM0MWNhNmRlOWIxNjU0NmRmOTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WRFO2WsuZ5QG2-SU8BLg2SW0eAs8gIvAWKiJ8xyfAOw)

yes ryan you can go ahead and close jasmine ticket
